src/app/demo/demo.component.ts

```diff
@@ -0,0 +1,72 @@
+import { Component, OnInit } from '@angular/core';
+import {
+  NgbModal,
+  NgbModalRef
+} from '@ng-bootstrap/ng-bootstrap';
+import { constants } from '../constants';
+
+
+@Component({
+  selector: 'app-demo',
+  templateUrl: './demo.component.html',
+  styleUrls: ['./demo.component.scss']
+})
+export class DemoComponent implements OnInit {
+  checkboxStatus: boolean;
+  confirmOverlay: boolean;
+  confirmVisible: boolean;
+  displayOverlay: boolean;
+  isCardFreeze: boolean;
+  private modalRef: NgbModalRef;
+  openModal: boolean;
+  freezeConfirm: string;
+  freezeNote: string;
+  lostStolen: string;
+  freezeCard: string;
+  labels = constants.labels.freeze;
+  displayLabel = false;
+  constructor(private modalService: NgbModal) { }
+  ngOnInit() {
+    this.displayOverlay = false;
+    this.confirmVisible = true;
+    this.confirmOverlay = false;
+    this.checkboxStatus = false;
+    this.openModal = true;
+    this.isCardFreeze = false;
+  }
+
+  openNew() {
+    this.displayOverlay = true;
+    if (this.isCardFreeze === false) {
+      this.freezeConfirm = this.labels.freezeConfirm;
```

Comment:
Don't use if else, keep the previous conditions.
Reduce the variables, assign multiple lines to a single variable.